upgrading/cns_upgrades.adoc

```diff
@@ -0,0 +1,85 @@
+[[upgrading-CNS]]
+= Upgrading CNS on {product-title}
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[special-considerations-for-cns]]
+== Special Considerations for CNS
+When upgrading CNS on {product-title}, you must upgrade the set of nodes
+where CNS pods are running.
+
+. Manage the daemonset:
++
+----
+$ oc delete ds glusterfs  --cascade=false
+----
+
+. Ensure consistency in the cluster before proceeding:
++
+----
+oc delete pod <gluster_pod>
+----
+
+. Run the following command on each volume to check the self-heal status:
++
+----
+$ gluster volume heal <volname> info
+----
+
+When not upgrading CNS at all in a {product-title} upgrade, `drain` (which does
+not work on daemonsets) and `unschedule` (the daemonset will still create or
+recreate the pod on the node as pending) are run without any consideration for
+what is running there.
+
+There is also the potential for someone to run an upgrade on multiple nodes at
+the same time, which causes issues if more than one are hosting CNS pods.
+
+Even if a serial upgrade is running, there is no guarantee sufficient
+time will be given for node A to complete all of its healing operations before
+node B is terminated. This could leave the cluster in a bad or unknown state.
+Therefore, the following procedure is recommended.
+
+[[upgrading-CNS]]
+== Upgrading CNS
+
+. Upgrade the control plane, the master nodes and etcd nodes.
+
+. Upgrade standard infra nodes (router, registry, logging and metrics).
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade standard nodes running application containers.
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade the CNS cluster one node at a time, commenting out other nodes.
```

Comment:
@mrobson @jarrpa What commands should be listed for this procedure? I would like to be as specific as possible.

Reply:
What do you mean by "commenting out other nodes"?

Reply:
This was pulled in from a comment from @mrobson here: https://bugzilla.redhat.com/show_bug.cgi?id=1543003#c2